### ISSUE TITLE
'Unspecified' fields on browser  

### ISSUE THREAD

**Issue by pnejad** [DataBiosphere/data-browser#901]
We noticed a lot of 'Unspecified' fields in the browser in places that had information previously. 

For example: 

Organ, organ_part, and disease are showing 'Unspecified' for the  [Human Hematopoietic Profiling dataset](https://data.humancellatlas.org/explore/projects/091cf39b-01bc-42e5-9437-f419a66c8a45).

Tagging @HumanCellAtlas/wranglers to list other issues we noticed.

**Comment by theathorn** [comment]
@hannes-ucsc @NoopDog Is this a metadata issue, indexing issue or data browser issue?

**Comment by zperova** [comment]
@pnejad @hannes-ucsc @NoopDog @theathorn 
This dataset uses cell lines so there is a `model_organ` which is displayed correctly in the Explore Data view (where all the projects are listed).
I bet the other datasets that also have `Unspecified` in the Organ and Organ parts are those with cell lines or organoids. We introduced the Model organ because this is a correct way to represent the data however we have not discussed what should be displayed in the Explore Project view for such datasets. I think this needs a different ticket - calling for a correct and harmonized representation of such datasets. Displaying the actual values for Organ and Organ parts that are in the metadata is not a sensible solution. Adding Model organ to the Explore Project view would help. I found related ticket: https://github.com/HumanCellAtlas/data-browser/issues/805

@hannes-ucsc could you please comment on what is the algorithm used by the Data Browser to populate Organ, Organ parts and Known diseases (Specimen) in the Explore Project view and in the Explore Data view?

More related tickets for reference here:
https://github.com/HumanCellAtlas/data-browser/issues/338
https://github.com/HumanCellAtlas/data-browser/issues/302

**Comment by pnejad** [comment]
@zperova  Good catch, I think that is why it does not appear in the main table!

The lab did purchase a cell line, but they did not differentiate them. So the specimen organ/organ_part is the same as the model_organ/model_organ_part.


Here is how it currently looks in the browser:

![Screen Shot 2019-09-27 at 8 02 09 AM](https://user-images.githubusercontent.com/36082251/65779755-390cd200-e0fd-11e9-83a6-0f47da976e14.png)
![Screen Shot 2019-09-27 at 8 03 52 AM](https://user-images.githubusercontent.com/36082251/65779862-68bbda00-e0fd-11e9-88f3-25b4b0514a29.png)

**Comment by hannes-ucsc** [comment]
> @hannes-ucsc could you please comment on what is the algorithm used by the Data Browser to populate Organ, Organ parts and Known diseases (Specimen) in the Explore Project view and in the Explore Data view?

`Known diseases (Specimen)` is populated from [`specimen_from_organism`](https://github.com/HumanCellAtlas/metadata-schema/blob/aa4b9eac5aab6caf0dd77fbb8c21308254954069/json_schema/type/biomaterial/specimen_from_organism.json#L68)

Azul also exposes the donor diseases from [`donor_organism.disease`](https://github.com/HumanCellAtlas/metadata-schema/blob/aa4b9eac5aab6caf0dd77fbb8c21308254954069/json_schema/type/biomaterial/donor_organism.json#L111) but the Data Browser does not display that. @NoopDog should be able to clarify as to if this was intentional and if so what the intention is.

Organ is populated from the `sample.organ` field. A sample is the biomaterial from which the cell suspension was created. It can be a `specimen_from_organism`, `organoid` or `cell_line`. The problem is that only `specimen_from_organism` actually has a `organ` property, so for the other two types, the result is `Unspecified`. 

Filtering by sample type `organoid` or `cell_line` illustrates the problem well.

https://data.humancellatlas.org/explore/projects?filter=%5B%7B%22facetName%22%3A%22sampleEntityType%22%2C%22terms%22%3A%5B%22cell_lines%22%2C%22organoids%22%5D%7D%5D

<img width="604" alt="Screen Shot 2019-09-27 at 10 20 15 PM" src="https://user-images.githubusercontent.com/5143256/65811944-030b3480-e175-11e9-8824-a480e1a8a1da.png">

Only one of the projects has a value other than _Unspecified_ for _Organ_ and that's because that project has samples that are specimens, meaning that some cell suspensions in that project were derived from specimens directly while other cell suspensions were derived from cell lines that were derived from specimens. We see the organ of the sample specimens but we don't see the organ of specimens that aren't samples.

TL;DR: The DB seems to compute the values in _Organ / Model Organ_ column as in `(samples[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified"`. One resolution would be to make that `(specimens[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified")`.

Organ Part is handled identically to organ. Everything I said about organ applies to organ part. 

You can stop reading now because I think the above solution will work except for freaky graphs: That solution does introduce the possibility of including the organ of specimens that aren't connected to a cell suspension. If that turns out to be a problem the solution in Azul would be to propagate `specimen.organ` into the connected sample. That would mean that organoids and cell suspension would have both `organ` and `modelOrgan` properties in the response. We've stopped pushing properties down like that because it created problems with aggregation by organ. Another solution would be to disregard specimens that aren't connected to a sample.

**Comment by theathorn** [comment]
Discuss with @hannes-ucsc

**Comment by NoopDog** [comment]
>Azul also exposes the donor diseases from donor_organism.disease but the Data Browser does not display that. @NoopDog should be able to clarify as to if this was intentional and if so what the intention is.

Correct, the donor_organism.disease is not currently displayed. We can display but this just needs to be clearly distinguished from the sample disease. (the donor may have a disease but the organ the sample was taken from was "normal")

**Comment by NoopDog** [comment]
Regarding:

>TL;DR: The DB seems to compute the values in Organ / Model Organ column as in (samples[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified". One resolution would be to make that (specimens[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified").

We have validated the Data Browser is working as originally designed.  I need to look a bit deeper into the TLDR to see which solution makes sense to me. Stay tuned.

**Comment by NoopDog** [comment]
Regarding the current state:

I have confirmed that:

>The DB is computing the values in Organ / Model Organ column as in (samples[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified".

as @hannes-ucsc figured.

Interestingly cell line and organoid metadata do not support the representation of the organ or organ part they were derived from. Cell line does have an optional "Tissue" with an organ part https://data.humancellatlas.org/metadata/dictionary/biomaterial/cell_line#cell_line-tissue but it seems this is not collected and the documentation specifies organ part and not organ. 

So in the current state, the organ and organ part will be unspecified in the data browser for any project without a specimen from organism.( that only uses cell lines or organoids)


Regarding the Proposed Solution to always take organ and organ part from specimen form organism:

>That solution does introduce the possibility of including the organ of specimens that aren't connected to a cell suspension

@hannes-ucsc how could this happen? This would just be a sample that was taken but never used but who's metadata is somehow included  the project?
.
Propagating specimen from organism down could work especially for the samples tab but this has the problem that it's more work for Azul and not really the direction we want to go. Also it introduces a change in the semantics of the field depending on where it is in the graph. On specimen it means thats the organ that was sampled. On cell line or organoid it means the organoid or cell line was derived from a specimen with the given organ type. We might want to make "collected from" and "derived from" more explicit but this would require metadata changes.

Regarding
>One resolution would be to make that (specimens[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified").

I can put this on ux-dev tomorrow  pointing to prod to let you see what it looks like. For the samples and files tabs this will show the organ used to derive the sample if one is included in the metadata graph.

For the projects tab, this will show all specimen form organism organs which is what we want. 

@zperova @pnejad do we have projects that have specimen from organism that are never used in the project? 

Will let you know when you can review this on ux-dev (pointing to prod data)

**Comment by pnejad** [comment]
@NoopDog In stem cell/organoid experiments the collected specimen will not be the same as the final "model organ", but we shouldn't have a specimen that is never used in the project.

**Comment by NoopDog** [comment]
OK ux-dev which points to prod data now has

https://ux-dev.data.humancellatlas.org/explore/projects?filter=%5B%7B%22facetName%22%3A%22genusSpecies%22%2C%22terms%22%3A%5B%22Homo+sapiens%22%5D%7D%2C%7B%22facetName%22%3A%22sampleEntityType%22%2C%22terms%22%3A%5B%22cell_lines%22%5D%7D%5D

> (specimens[].organ or "Unspecified") + "/" + (samples[].modelOrgan or "Unspecified").
@zperova @pnejad is this matching what you are expecting to see?

Note that "disease" is still coming from samples, should disease come from specimens as well @hannes-ucsc ?

**Comment by hannes-ucsc** [comment]
> Note that "disease" is still coming from samples, should disease come from specimens as well @hannes-ucsc ?

Yes. While you're at it, can we overload that column to include donor disease as well? The column header could be 

```
Donor disease /
Specimen disease
```

Semantically this is similar to organ vs model organ: something from higher in the graph followed by slash followed by something from lower in the graph.

I don't think the word "status" in the column header adds much.

**Comment by hannes-ucsc** [comment]
@pnejad, @zperova does ux-dev (which @NoopDog links to above) look good? To me it does.

**Comment by pnejad** [comment]
Looks good to me!

**Comment by lauraclarke** [comment]
> > Note that "disease" is still coming from samples, should disease come from specimens as well @hannes-ucsc ?
> 
> Yes. While you're at it, can we overload that column to include donor disease as well? The column header could be
> 
> ```
> Donor disease /
> Specimen disease
> ```
> 
> Semantically this is similar to organ vs model organ: something from higher in the graph followed by slash followed by something from lower in the graph.
> 
> I don't think the word "status" in the column header adds much.

The purpose of disease status is when it says normal, it isn't a disease so the lack of the word status makes no sense.

**Comment by diekhans** [comment]
If the field is defined as "disease", "none" would be better that "normal".  This
seems semantically better if we are storing the actually disease in there.

**Comment by lauraclarke** [comment]
Normal is used because we aren't asserting for definite that the person does not have any disease but we are asserting they appear normal on current medical scales, normal being the acceptable phenotype term for that https://www.ebi.ac.uk/ols/ontologies/pato/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FPATO_0000461

**Comment by diekhans** [comment]
normal sounds great then!

**Comment by NoopDog** [comment]
@hannes-ucsc With the above change I think we need to have Azul populate the term facets from specimen.organ an specimen.organ part (instead of samples) as well.  Would you like me to create a ticket for that? I will revert my changes until the Azul ticket comes through.  Thats right, right?

**Comment by hannes-ucsc** [comment]
@NoopDog 

> With the above change I think we need to have Azul populate the term facets from specimen.organ an specimen.organ part (instead of samples) as well. Would you like me to create a ticket for that?

https://github.com/DataBiosphere/azul/issues/1497

> I will revert my changes until the Azul ticket comes through. Thats right, right?

Right.

**Comment by zperova** [comment]
@NoopDog ux-dev pointing at prod data looks good.
Could you please change to 'Donor disease status/Specimen disease status' as per @lauraclarke explanation.
